code/modules/atmospherics/machinery/portable/canister.dm

```diff
@@ -482,12 +493,30 @@
 		valve_open = !valve_open
 		timing = FALSE
 
-	// Handle gas transfer.
+	var/turf/location = get_turf(src)
+
+	var/mix_air = FALSE
+	var/pressure = release_pressure
+	var/gas_mix = holding?.air_contents
+	var/air_update = FALSE
+
 	if(valve_open)
-		var/turf/T = get_turf(src)
-		var/datum/gas_mixture/target_air = holding ? holding.air_contents : T.return_air()
+		mix_air = TRUE
+
+	// When at least 10% of integrity is lost it starts checking for leaking
+	if(obj_integrity < max_integrity * 0.9)
+		var/leak_chance = (1 - obj_integrity / max_integrity) * 100
+		if(prob(leak_chance))
+			mix_air = TRUE
+			pressure = air_contents.return_pressure() / 10
```

Comment:
Shouldn't we max this and the release pressure?

Reply:
Release pressure is what gets set in the ui by the player, and isn't used by leaks. Leakages are basically a crack in the canister and are intentionally slowed down.

Reply:
yeah but if the release pressure is higher then the pressure of the mix / 10, shouldn't we use that?

Reply:
ah I see, if the release valve is open and there's a hole in the tank you want to use the hole's pressure each time